### PR TITLE
add whatLanguage command

### LINKED ISSUE #1179: LanguageLayer "whatlanguage" Command
`s!whatlanguage <text>` - Tells you what language a piece of text is in

https://languagelayer.com/

### PR DESCRIPTION
**Pull request by Vicente015**
Hey, I have created the command based on your code and looking for how things work. I've been wanting to contribute to this project for a long time, so please let me know if there's something wrong. I'm not sure if the changes I have made work, but I try to do my best. 
This closes #1179

### CODE REVIEW

File: src/apis/LanguageLayer.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const { APIWrapper } = require('../')
+const fetch = require('node-fetch')
+
+const API_URL = 'http://api.languagelayer.com/'
+
+module.exports = class TumblrAPI extends APIWrapper {
```
Comment (almeidx) [inline src/apis/LanguageLayer.js:6]: Tubmlr?

Comment (Vicente015) [reply src/apis/LanguageLayer.js:6]: shit, I forgot to remove that, sorry

Comment (Vicente015) [reply src/apis/LanguageLayer.js]: I have already fixed it, tell me if there is something else 😓

**Review by almeidx — COMMENTED** [review]
Check the *checks* tab, your code style does not conform to our standard

**Comment by Vicente015** [comment]
fixed :)

**Comment by Vicente015** [comment]
It is necessary to add `LANGUAGELAYER_API_KEY` to .env.example? I think that may be missing

**Comment by almeidx** [comment]
> It is necessary to add `LANGUAGELAYER_API_KEY` to .env.example? I think that may be missing

That and to the Environment Variables wiki page

**Comment by Vicente015** [comment]
Ok, I'll do it, thanks

**Comment by Vicente015** [comment]
> > It is necessary to add `LANGUAGELAYER_API_KEY` to .env.example? I think that may be missing
> 
> That and to the Environment Variables wiki page

Done!

**Review by almeidx — DISMISSED** [review]
LGTM

**Review by almeidx — COMMENTED** [review]
Btw, could you use [axios](https://www.npmjs.com/package/axios) instead of node-fetch? As per #1096

**Comment by Vicente015** [comment]
> Btw, could you use [axios](https://www.npmjs.com/package/axios) instead of node-fetch? As per #1096

Done i think

File: src/apis/LanguageLayer.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  detectText (params = {}) {
+    return this.request(`/detect`, `?${params}`)
+  }
+
+  request (endpoint, query = {}) {
+    let apikey = `&access_key=${process.env.LANGUAGELAYER_API_KEY}`
+    return axios.get(API_URL + endpoint + apikey + query)
+      .then(res => res.json())
```
Comment (almeidx) [inline src/apis/LanguageLayer.js:20]:
```suggestion
```

File: src/commands/misc/whatlanguage.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  async run ({ t, author, channel }, query) {
+    channel.startTyping()
+    const result = await this.client.apis.languagelayer.detectText(query)
+    const embed = new SwitchbladeEmbed(author)
+      .setTitle(t('commands:whatlanguage.embedTitle'))
+      .setDescription(t('commands:whatlanguage.embedDes'))
+      .addField(t('commands:whatlanguage.fieldLanguage'), `:flag_${result.results[0].language_code}: ${result.results[0].language_name}`, true)
+      .addField(t('commands:whatlanguage.fieldProbability'), `${Math.round(result.results[0].probability)}%`, true)
+      .addField(t('commands:whatlanguage.fieldPercentage'), `${result.results[0].percentage}%`, true)
```
Comment (almeidx) [inline src/commands/misc/whatlanguage.js:22]:
```suggestion
    const { data } = await this.client.apis.languagelayer.detectText(query)
    const embed = new SwitchbladeEmbed(author)
      .setTitle(t('commands:whatlanguage.embedTitle'))
      .setDescription(t('commands:whatlanguage.embedDes'))
      .addField(t('commands:whatlanguage.fieldLanguage'), `:flag_${data.results[0].language_code}: ${data.results[0].language_name}`, true)
      .addField(t('commands:whatlanguage.fieldProbability'), `${Math.round(data.results[0].probability)}%`, true)
      .addField(t('commands:whatlanguage.fieldPercentage'), `${data.results[0].percentage}%`, true)
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: src/apis/LanguageLayer.js (hunk 2 above)
Comment (Vicente015) [reply src/apis/LanguageLayer.js:20]: Could you explain why? @almeidx

Comment (almeidx) [reply src/apis/LanguageLayer.js]: As stated in the issue I linked earlier:
> Axios performs automatic transformations of JSON data.

Comment (Vicente015) [reply src/apis/LanguageLayer.js]: Oh okay, thanks :)

**Review by almeidx — COMMENTED** [review]
In theory, it should all still work the same, but please test it

File: src/apis/LanguageLayer.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  detectText (params = {}) {
+    return this.request(`/detect`, `?${params}`)
+  }
+
+  request (endpoint, query = {}) {
+    let apikey = `&access_key=${process.env.LANGUAGELAYER_API_KEY}`
+    return axios.get(API_URL + endpoint + apikey + query)
+  }
+}
```
Comment (almeidx) [inline src/apis/LanguageLayer.js:21]:
```suggestion
const { APIWrapper } = require('../')
const axios = require('axios')

const API_URL = 'http://api.languagelayer.com'

module.exports = class LanguageLayerAPI extends APIWrapper {
  constructor () {
    super({
      name: 'languagelayer',
      envVars: ['LANGUAGELAYER_API_KEY']
    })
  }

  detectText (query) {
    return axios(`${API_URL}/detect`, {
      params: {
        access_key: process.env.LANGUAGELAYER_API_KEY,
        query
      }
    })
  }
}

```

**Review by almeidx — CHANGES_REQUESTED** [review]
Fixed some issues

**Review by Doges — APPROVED** [review]
LGTM

**Pull request merged**